### PR TITLE
support FOAMCUTTER_XYUV with RAMPS

### PR DESCRIPTION
**Pull request by DerAndere1**
### Description

Modify pins_RAMPS.h to properly support 4 axis foam cutters.  An alternative would be to create a new board variant.

### Requirements

RAMPS board, add `#define FOAMCUTTER_XYUV` in Configuration.h

### Benefits

Allow compilation of an example config for a 4 axis foam cutter (to be added in PR https://github.com/MarlinFirmware/Configurations/pull/751)

### Configurations

https://github.com/DerAndere1/Configurations/tree/FOAMCUTTER_XYUV_by_rcKeith/config/examples/FoamCutter/rcKeith/Foam_Cutter

### Related Issues

https://github.com/MarlinFirmware/Configurations/pull/751

### CODE REVIEW

**Comment by thinkyhead** [comment]
If the foamcutter only needs one endstop per axis, then it only needs to define a `*_STOP_PIN` for each of the axes, and then it doesn't need to have both a MIN and a MAX endstop plug defined, but just a preferred pin. For most RAMPS boards, it's sensible to have both MIN and MAX endstops defined for XYZ so the actual plugs match up, but for I, J, K… it makes more sense to just pick one STOP pin and always prefer that, saving on the need to move the pin if it's MAX versus MIN.

**Comment by DerAndere1** [comment]
Thanks for putting some extra thought into this. I am beginning to understand how to use the MAP macro.

**Comment by thinkyhead** [comment]
> I am beginning to understand how to use the `MAP` macro.

Cool! On that reminder, I've added more info about Marlin macros to the website.

File: Marlin/src/pins/ramps/pins_RAMPS.h [hunk 1]
```diff
@@ -78,6 +78,31 @@
...
+
+#if ENABLED(FOAMCUTTER_XYUV)
+  #ifndef MOSFET_C_PIN
+    #define MOSFET_C_PIN                      -1
+  #endif
+  #if HAS_CUTTER && !defined(SPINDLE_LASER_ENA_PIN) && NUM_SERVOS < 2
+    #define SPINDLE_LASER_PWM_PIN              8  // Hardware PWM
+  #endif
+  #ifndef Z_MIN_PIN
+    #define Z_MIN_PIN                         -1
```
Comment (thisiskeithb) [inline Marlin/src/pins/ramps/pins_RAMPS.h:93]: @DerAndere1: How do you home Z on foam cutters? I've found some online with Y & Z endstops, but you're unsetting the Z endstop here when one isn't defined and that causes compiling to fail for both of our [FoamCutter configs](https://github.com/MarlinFirmware/Configurations/tree/bugfix-2.1.x/config/examples/FoamCutter) with a `Z_MIN_PIN, Z_STOP_PIN, or Z_SPI_SENSORLESS is required for Z axis homing` error.

**Pull request merged**